tensorflow_probability/python/experimental/mcmc/sample_discarding_kernel.py

```diff
@@ -0,0 +1,178 @@
+"""Sample Discarding Kernel for Thinning and Burn-in"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+
+# Dependency imports
+import tensorflow.compat.v2 as tf
+from tensorflow_probability.python.experimental.mcmc import step_kernel
+from tensorflow_probability.python.mcmc import kernel as kernel_base
+from tensorflow_probability.python.mcmc.internal import util as mcmc_util
+
+
+__all__ = [
+    'SampleDiscardingKernel',
+]
+
+
+class SampleDiscardingKernelResults(
+    mcmc_util.PrettyNamedTupleMixin,
+    collections.namedtuple('SampleDiscardingKernelResults',
+                           ['call_counter',
+                            'inner_results'])):
+  __slots__ = ()
+
+
+class SampleDiscardingKernel(kernel_base.TransitionKernel):
+  """Appropriately discards samples to conduct thinning and burn-in.
+
+  `SampleDiscardingKernel` is a composable `TransitionKernel` that
+  applies thinning and burn-in to samples returned by its
+  `inner_kernel`. All Transition Kernels wrapping it will only
+  see non-discarded samples.
+
+  The burn-in step conducts both burn-in and one step of thinning.
+  In other words, the first call to `one_step` will skip
+  `num_burnin_steps + num_steps_between_results` samples. All
+  subsequent calls skip only `num_steps_between_results` samples.
+  """
+
+  def __init__(
+      self,
+      inner_kernel,
+      num_burnin_steps=0,
+      num_steps_between_results=0,
+      name=None):
+    """Instantiates this object.
+
+    Args:
+      inner_kernel: `TransitionKernel` whose `one_step` will generate
+        MCMC sample(s).
+      num_burnin_steps: Integer number of chain steps to take before starting to
+        collect results. Defaults to 0 (i.e., no burn-in).
+      num_steps_between_results: Integer number of chain steps between
+        collecting a result. Only one out of every
+        `num_steps_between_samples + 1` steps is included in the returned
+        results. Defaults to 0 (i.e., no thinning).
+      name: Python `str` name prefixed to Ops created by this function.
+        Default value: `None` (i.e., "sample_discarding_kernel").
+    """
+    if tf.get_static_value(num_burnin_steps):
+      num_burnin_steps = tf.get_static_value(num_burnin_steps)
+    if tf.get_static_value(num_steps_between_results):
+      num_steps_between_results = tf.get_static_value(
+          num_steps_between_results)
+    self._parameters = dict(
+        inner_kernel=inner_kernel,
+        num_burnin_steps=num_burnin_steps,
+        num_steps_between_results=num_steps_between_results,
+        name=name or 'sample_discarding_kernel'
+    )
+
+  def _num_samples_to_skip(self, call_counter):
+    """Calculates how many samples to skip based on the call number.
+
+    If `self.num_burnin_steps` is statically known to be 0,
+    `self.num_steps_between_results` will be returned outright.
```

Comment:
Google docstring style: If the docstring is longer than one line, then an Args: and a Returns: section are required.

In this case, since the method is private, you could choose to shorten the docstring to just the summary, and make the rest a code comment.